lib/std/mem.zig

```diff
@@ -273,6 +273,33 @@ pub fn set(comptime T: type, dest: []T, value: T) void {
         d.* = value;
 }
 
+/// Zero initializes the type.
+/// This can be used to zero initialize a C-struct.
+pub fn zeroes(comptime T: type) T {
+    if (@sizeOf(T) == 0) return T{};
+
+    if (comptime meta.containerLayout(T) != .Extern) {
+        @compileError("TODO: Currently this only works for extern types");
+    }
+
+    var item: T = undefined;
+    @memset(@ptrCast([*]u8, &item), 0, @sizeOf(T));
```

Comment:
Is this defined? e.g. if `T` was a `u24` then the size is 3 but `@sizeOf` returns 4.
See e.g. #4093

Reply:
The code is correct. `@sizeOf` gives the number of bytes the type takes up in memory. This function is valid for types which have well-defined memory layout, which includes extern structs (the main use case).